UnresolvedMemberLookupTests.cpp

```diff
@@ -0,0 +1,82 @@
+//===--- ConstraintSimplificationTests.cpp --------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#include "SemaFixture.h"
+#include "swift/Sema/ConstraintSystem.h"
+
+using namespace swift;
+using namespace swift::unittest;
+using namespace swift::constraints;
+
+TEST_F(SemaTest, TestLookupAlwaysLooksThroughOptionalBase) {
+  auto *intTypeDecl = getStdlibNomicalTypeDecl("Int");
+  auto *optTypeDecl = getStdlibNomicalTypeDecl("Optional");
+  auto intType = intTypeDecl->getDeclaredType();
+  auto intOptType = OptionalType::get(intType);
+  auto stringType = getStdlibType("String");
+
+  auto *intMember = addExtensionVarMember(intTypeDecl, "test", intOptType);
+  addExtensionVarMember(optTypeDecl, "test", stringType);
+
+  auto *UME = new (Context) UnresolvedMemberExpr(SourceLoc(), DeclNameLoc(), DeclNameRef(Context.getIdentifier("test")),
```

Comment:
Nit: Should this be clang-formatted? Seems like it's not in the 80 col rule :)

Reply:
You are correct :)